### PR TITLE
Users/hdo/lesson1 2

### PR DESCRIPTION
**Pull request by HuongDoViedoc**
2 first lessons.

### CODE REVIEW

File: BasicTraining/Program.cs [hunk 1]
```diff
@@ -1,2 +1,58 @@
...
+    else
+    {
+        try
+        {
+            result = sss.divide(number1, number2);
+            result = System.Math.Round(result, 2);
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine("Cannot divide by 0");
```
Comment (lavender1289) [inline BasicTraining/Program.cs]: This program never catches this exception

File: BasicTraining/Program.cs [hunk 2]
```diff
@@ -1,2 +1,58 @@
...
+        }
+    }
+if(number2 != 0)
+    Console.WriteLine("Result of "+sign+" 2 numbers is: " + result);
```
Comment (lavender1289) [inline BasicTraining/Program.cs]: If number2 = 0 and operator is add, subtract, multiply, this result never be printed

File: BasicTraining/Program.cs (hunk 1 above)
Comment (HuongDoViedoc) [reply BasicTraining/Program.cs]: Ok. resolved.

File: BasicTraining/Program.cs (hunk 2 above)
Comment (HuongDoViedoc) [reply BasicTraining/Program.cs]: yeap. resolved.

**Comment by viedoc-trungle** [comment]
There are 3 main problems with your code:
- A lot of redundant code, it's harder to main this way.
- Most of the program concerns are still stay on `Main` method. Let's list it:
    - Get inputs from user interaction(multiple times)
    - Parse inputs(multiple times)
    - Get/parse inputs again from user interaction if the previous inputs are incorrect(multiple times)
    - Validate inputs for divide operator
    - Calculate
    - Format result
    - Show result
- Code flow is still not clear enough. To catch the intention of the whole program, one should read through 40 lines of code. I would expect 2-3 lines of code for this program. We will have a lot more complicated programs than this.

**Comment by viedoc-trungle** [comment]
LGTM

**Review by viedoc-trungle — APPROVED** [review]
LGTM

**Pull request merged**